agreement/asyncVoteVerifier.go

```diff
@@ -140,16 +140,18 @@ func (avv *AsyncVoteVerifier) verifyVote(verctx context.Context, l LedgerReader,
 		// if we're done while waiting for room in the requests channel, don't queue the request
 		req := asyncVerifyVoteRequest{ctx: verctx, l: l, uv: &uv, index: index, message: message, out: out}
 		avv.wg.Add(1)
-		if avv.backlogExecPool.EnqueueBacklog(avv.ctx, avv.executeVoteVerification, req, avv.execpoolOut) != nil {
+		if err := avv.backlogExecPool.EnqueueBacklog(avv.ctx, avv.executeVoteVerification, req, avv.execpoolOut); err != nil {
```

Comment:
Interestingly, inside of EnqueueBacklog it is also doing a select on `case <-avv.ctx.Done()` (passed in as enqueueCtx) which will cause EnqueueBacklog to return ctx.Err(). If you have two pending selects like that on the stack, is it deterministic which one will fire first?

Reply:
There is this comment warning "instead, enqueue so the worker will set the error value and return the cancelled vote properly." However if the waiting inside EnqueueBacklog() is interrupted by one of the Done()s it's waiting on, it will never reach the worker.

Reply:
The fact that we have two pending selects isn't an issue. The one in `verifyVote` would be evaluated when we get to that point, and if it's not canceled yet, it would go to the `default` statement, executing `EnqueueBacklog`, where it would also block on the same channel.

As for the comment, I think it's a bug - I think that the intent was to have a `fallthrough` here. But I would leave that to a separate PR, as it's not really related to this change. ( i.e. in out case, we're passing a TODO context, which would never expire and therefore this statement would never be executed )

Reply:
there are other callers (unauthenticatedBundle.verify and cryptoVerifier.voteFillWorker) passing a valid context into verifyVote — just not pseudonode

Reply:
yes, I'm aware of that - that's why I did not attempted to change this code ;-)